daffodil-io/src/main/scala/edu/illinois/ncsa/daffodil/processors/charset/CharsetUtils.scala

```diff
@@ -52,21 +49,29 @@ import edu.illinois.ncsa.daffodil.util.MaybeInt
 
 /**
  * Serializable Charset
+ *
+ * java Charset are not serializable, but we need them to be.
+ * So this is serialized as a charset name, and fills in the charset lazily on
+ * first use.
+ *
+ * It's important that this contains a charset, but is not itself a charset.
+ * That way we can't mistake a charset (not serializable) for one of these.
  */
-case class DFDLCharset(val charsetName: String) extends Serializable {
-  import java.nio.charset.StandardCharsets
-
+case class DFDLCharset(_initialName: String) extends Serializable {
   charset // Force charset to be evaluted to ensure it's valid at compile time.  It's a lazy val so it will be evaluated when de-serialized
-  @transient lazy val charset = CharsetUtils.getCharset(charsetName)
+  @transient lazy val charset = CharsetUtils.getCharset(_initialName)
   @transient lazy val maybeFixedWidth = CharsetUtils.maybeEncodingFixedWidth(charset)
 
+  def charsetName = charset.name
+  def name = charset.name
```

Comment:
Can one of these functions be removed, or do they serve different purposes that an overriding class might change?